logprep/framework/rule_tree/node.py

```diff
@@ -9,7 +9,13 @@
 class Node:
     """Tree node for rule tree model."""
 
-    def __init__(self, expression: FilterExpression):
+    __slots__ = ("_expression", "_children", "matching_rules")
+
+    _expression: Union[FilterExpression, str]
+    _children: list
```

Comment:
if we can achieve to make the FilterExpression class hashable, we could use a dict here with the filterexpression as key. this ensures uniqueness and accessing or lookup an element in a dict has O(1) complexity. This could be an improvement for get_matching_rules in future refactorings

Reply:
Do I undertand it correctly that you would like to use a dict for the "matching_rules" attribute later on, but for now just make the FIlterExpressions hashable?

Reply:
we could do both in future itereations